core/kotlinx-coroutines-core/src/Executors.kt

```diff
@@ -10,43 +10,37 @@ import java.util.concurrent.*
 import kotlin.coroutines.experimental.*
 
 /**
- * [CoroutineDispatcher] that implements [Closeable]
+ * [CoroutineDispatcher] that has underlying [Executor] for dispatching tasks.
+ * Instances of [ExecutorCoroutineDispatcher] should be closed by the owner of the dispatcher.
+ *
+ * This class is generally used as a bridge between coroutine-based API and
+ * asynchronous API which requires instance of the [Executor]
  */
-abstract class CloseableCoroutineDispatcher: CoroutineDispatcher(), Closeable
+public abstract class ExecutorCoroutineDispatcher: CoroutineDispatcher(), Closeable {
```

Comment:
I'd suggest to keep maintaining binary compatibility with previously compiled code by using `@JvmName("CloseableCoroutineDispatcher")` to this declaration and adding a deprecated `typealias CloseableCoroutineDispatcher = ExecutorCoroutineDispatcher` with an appropriate `replaceWith`. We will clean it up in 1.0 release.